compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -466,7 +466,10 @@ class TreePickler(pickler: TastyPickler, attributes: Attributes) {
               }
             case _ =>
               if passesConditionForErroringBestEffortCode(tree.hasType) then
-                val sig = tree.tpe.signature
+                // #19951 The signature of a constructor of a Java annotation is irrelevant
+                val sig =
+                  if name == nme.CONSTRUCTOR && tree.symbol.owner.is(JavaAnnotation) then Signature.NotAMethod
```

Comment:
The change to signatures looks like the only part that couldn't be backported to 3.3 (because it would break the existing unpickler logic). If we agree that we'd like to backport the rest to 3.3, maybe this change should be done in a separate PR after we've validated that the other changes work together?

Reply:
If we leave this out, we also need to leave out the other change in this file. Otherwise we'll pickle constructor calls where the actual arguments do not match the signature, so an older compiler will also choke on it.

That said, as the latest push shows, even explicitly expanding all `NamedArg`s (the first commit) will break the current version of WartRemover. So I don't think we can backport this to 3.3.x at all.

Reply:
I guess this will need to be discussed more, but I think breaking WartRemover could be OK if we allow enough time for upstream to release a fixed version before we release a 3.3.x with the backport in. It seems like this would lead to less issues in practice in the long term.